### ISSUE TITLE
Invalid key provided

### ISSUE THREAD

**Issue by Ficik** [peers/peerjs#439]
Provided `options.key` is ignored and overwritten with `'peerjs'`:

https://github.com/peers/peerjs/blob/b88da6a1951802d1319e3b286e2e5105b5cd746b/lib/peer.js#L36

**Comment by kidandcat** [comment]
Yes, right now it is not used, I'm just overwritting it waiting for a full refactor of the library.

Key option was used to setup a PeerServer cloud service by giving unique keys to users. Anyway it is not a good system because the key is stored on client side and so anybody can get and use it.

I just overwrite it to avoid compatibility issues.

**Comment by Ficik** [comment]
I'm hosting own peerjs-server with different key. I want to use non-default key since server is accessible from internet. But apps are for internal use only so I don't care much about keys on client side.

**Comment by kidandcat** [comment]
I would recommend you to use a proxy and a custom header, that would be more secure, but if you want to use the key option, you can clone the repository, comment that line which overrides the option.key, and execute "grunt", then you will have your own build of peer.js in dist folder.

**Comment by kidandcat** [comment]
An older release may not work with latest browser versions. Creating you own build is easy. Anyway, if you need it that much (instead of managing authentication outside of PeerJS using headers and proxies), I can remove the key overwrite.
